contracts/Faucet.sol

```diff
@@ -9,113 +9,105 @@ import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
     function setToucanContractRegistry(
-        address _address
+        address contractRegistry_
+    ) public virtual onlyOwner {
+        contractRegistry = contractRegistry_;
+    }
+
+    /// @notice allows the owner to set the eligibility of a pool
+    /// @param poolAddress the address of the pool
+    /// @param isPoolEligible_ eligibility flag
+    function setPoolEligible(
+        address poolAddress,
+        bool isPoolEligible_
     ) public virtual onlyOwner {
-        contractRegistry = _address;
+        isPoolEligible[poolAddress] = isPoolEligible_;
     }
 
     /// @notice A function to get the Faucet's balances of multiple tokens at once
-    /// @param _erc20Addresses An array of ERC20 contract addresses
+    /// @param erc20Addresses An array of ERC20 contract addresses
     /// @return An array of balances
     function getTokenBalances(
-        address[] memory _erc20Addresses
+        address[] memory erc20Addresses
     ) public view returns (uint256[] memory) {
-        uint256[] memory balances = new uint256[](_erc20Addresses.length);
-        for (uint256 i = 0; i < _erc20Addresses.length; i++) {
-            balances[i] = IERC20(_erc20Addresses[i]).balanceOf(address(this));
+        uint256[] memory balances = new uint256[](erc20Addresses.length);
+        for (uint256 i = 0; i < erc20Addresses.length; i++) {
+            balances[i] = IERC20(erc20Addresses[i]).balanceOf(address(this));
         }
         return balances;
     }
 
-    /// @notice checks if token to be deposited is eligible for the Faucet
-    /// @param _erc20Address address to be checked
-    function checkTokenEligibility(
-        address _erc20Address
-    ) private view returns (bool) {
-        bool isToucanContract = IToucanContractRegistry(contractRegistry)
-            .checkERC20(_erc20Address);
-        if (isToucanContract) return true;
-
-        if (_erc20Address == bctAddress) return true;
-
-        if (_erc20Address == nctAddress) return true;
-
-        return false;
-    }
-
     /// @notice deposit tokens from caller to Faucet
-    /// @param _erc20Address ERC20 contract address to be deposited
-    /// @param _amount amount to be deposited
-    function deposit(address _erc20Address, uint256 _amount) public {
-        bool eligibility = checkTokenEligibility(_erc20Address);
-        require(eligibility, "Token rejected");
+    /// @param erc20Address ERC20 contract address to be deposited
+    /// @param amount amount to be deposited
+    function deposit(address erc20Address, uint256 amount) public {
+        require(_checkEligible(erc20Address), "Token rejected");
 
-        IERC20(_erc20Address).safeTransferFrom(
+        IERC20(erc20Address).safeTransferFrom(
             msg.sender,
             address(this),
-            _amount
+            amount
         );
 
-        emit Deposited(_erc20Address, _amount);
+        emit Deposited(erc20Address, amount);
```

Comment:
I think these events are worthless because anyone can transfer tokens directly to the contract and these transfers won't be captured in these events. Something to think about, we should likely drop them in a follow-up. We should probably drop `deposit` altogether and simply update the faucet UI to transfer directly from the token contract.

Reply:
what are these events currently used for?

Reply:
I don't think they are used anywhere